caikit/config/config.py

```diff
@@ -136,19 +150,39 @@ def merge_configs(base: Optional[dict], overrides: Optional[dict]) -> dict:
     if overrides is None:
         return base or {}
 
+    if merge_strategy == "override":
+        base.update(overrides)
+        return base
+
     # Do the deep merge
     for key, value in overrides.items():
         if (
             key not in base
-            or not isinstance(base[key], dict)
-            or not isinstance(value, dict)
+            or not isinstance(base[key], (dict, list))
+            or not isinstance(value, (dict, list))
         ):
             base[key] = value
+        elif isinstance(value, list):
+            if key not in base or not isinstance(base[key], list):
+                base[key] = value
+            else:
+                base[key] = merge_list(base[key], value)
         else:
-            base[key] = merge_configs(base[key], value)
+            base[key] = merge_configs(base[key], value, merge_strategy)
 
     return base
 
 
+def merge_list(base_list: list, new_list: list) -> list:
+    for val in new_list:
```

Comment:
ahhh right I did this loop with equality checks because dicts aren't hashable so you can't do anything fancy like this which would preserve order:
```
unique_list = list(dict.fromkeys(my_list))
```

Reply:
>But for the things that we put in lists in caikit config I don't know if we need to?

🤔 Yeah, I guess that is true, though I had thought we _might_ want duplicate entries in `load_priority`/`train_priority`. I guess my bigger point is that this seems like _extra_ code to prevent one possible problem (duplicates entries) which also potentially precludes a valid usecase. Why not just do the simpler thing of allowing duplicates and making that the user's problem?